lib/itk/queue/connection.ex

```diff
@@ -61,33 +85,29 @@ defmodule ITKQueue.Connection do
     [host: host, port: port, username: username, password: password, heartbeat: heartbeat]
   end
 
+  # This will block the GenServer until connection is opened successfully.
   defp connect(params) do
-    AMQP.Connection.open(params)
+    params
+    |> AMQP.Connection.open()
     |> handle_connection_result(params)
   end
 
-  defp handle_connection_result({:ok, connection}, _params) do
-    Process.link(connection.pid)
-    connection
+  defp handle_connection_result({:ok, connection}, params) do
+    Process.monitor(connection.pid)
+    %__MODULE__{params: params, connection: connection}
   end
 
   defp handle_connection_result({:error, _}, params) do
     Logger.info("AMQP connection failed, retrying")
-    Process.sleep(1000)
+    Process.sleep(@reconnect_interval)
```

Comment:
I was planning to remove the `sleep` since it'll block the current GenServer, as you already added `:reconnect` receiver, what do you think about using it with `Process.send(..)` after the `reconnect_interval` instead?

Reply:
This process can be called at any time for a connection, when we return invalid connection in between reconnect attempts, the connection pool will have to be updated to handle this by adding retry, and we need a put a limit to that retry, also need to check all places that always expect a valid connection. It would be nice to not block here but it just requires too many changes.

Reply:
Ok makes sense, I tried to change it before as well and it also blew up some tests.